src/control/server/storage/scm/provider.go

```diff
@@ -165,9 +165,8 @@ func (dsp *defaultSystemProvider) Mkfs(fsType, device string, force bool) error
 		// disable uninit_bg to initialize everything upfront
 		// disable resize to avoid GDT block allocations
 		// disable extra isize since we really have no use for this
-		// disable csum since we have ECC already for SCM
 		// bigalloc is intentionally not used since some kernels don't support it
-		"-O", "flex_bg,^uninit_bg,^resize_inode,^extra_isize,^metadata_csum",
```

Comment:
Is this potentially going to have a negative performance impact on kernels where it is supported (e.g. suse/leap 15 and i assume el8)?

Reply:
no, actually the other way around. Those features add overhead that we won't need.

Reply:
I was under the impression that the `^metadata_csum` notation meant that you were disabling a feature that is enabled by default. So my question is: Will that feature be enabled by default on newer kernels?

Reply:
Yes, this checksum feature will be enabled by default on recent kernels/e2fsprogs. As you know, i had to remove the "^metadata_csum" since older e2fsprogs don't understand it. This feature is also enabled in today's code on el8/leap15.
It does not incur any performance penalty, but just consumes a bit of space to store the checksums while we don't need it since we have ECC on the pmem.
To make a long story short, i think it would be nice to disable it, but that's not at all showstopper.
Recent kernels also support the spare_super2 option (not supported on EL7) that allows to reduce the overhead further. That being said, i did not enable it in this patch because of EL7 again ...

Reply:
damn it ... packed_meta_blocks isn't supported on el7 too ... it sounds like we are going to have to pass different options depending on the distribution :(

Reply:
OK, thanks for the explanation. I guess if it turns into a more important problem to solve eventually we could make a helper that returns platform-specific options.

Reply:
Oops, I guess our comments crossed mid-flight...

Reply:
> 
> 
> damn it ... packed_meta_blocks isn't supported on el7 too ... it sounds like we are going to have to pass different options depending on the distribution :(

I pushed #6466 which might help with that problem.